src/TimeSteppers/time_stepping_kernels.jl

```diff
@@ -209,14 +209,16 @@ function calculate_boundary_tendency_contributions!(Gⁿ, arch, U, C, args...)
 
     # Velocity fields
     for i in 1:3
-        apply_z_bcs!(Gⁿ[i], U[i], arch, args...)
+        apply_x_bcs!(Gⁿ[i], U[i], arch, args...)
```

Comment:
@ali-ramadhan this line is new in the algorithm; though for your benchmark they shouldn't do anything.